src/main/java/uk/gov/hmcts/reform/wataskconfigurationapi/auth/role/RoleAssignmentService.java

```diff
@@ -0,0 +1,74 @@
+package uk.gov.hmcts.reform.wataskconfigurationapi.auth.role;
+
+import feign.FeignException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.authorisation.generators.AuthTokenGenerator;
+import uk.gov.hmcts.reform.wataskconfigurationapi.auth.idam.IdamSystemTokenGenerator;
+import uk.gov.hmcts.reform.wataskconfigurationapi.auth.role.entities.Attributes;
+import uk.gov.hmcts.reform.wataskconfigurationapi.auth.role.entities.QueryRequest;
+import uk.gov.hmcts.reform.wataskconfigurationapi.auth.role.entities.RoleAssignment;
+import uk.gov.hmcts.reform.wataskconfigurationapi.auth.role.entities.RoleAssignmentResource;
+import uk.gov.hmcts.reform.wataskconfigurationapi.auth.role.entities.RoleName;
+import uk.gov.hmcts.reform.wataskconfigurationapi.auth.role.entities.RoleType;
+import uk.gov.hmcts.reform.wataskconfigurationapi.clients.RoleAssignmentServiceApi;
+import uk.gov.hmcts.reform.wataskconfigurationapi.exceptions.ServerErrorException;
+
+import java.time.LocalDateTime;
+import java.util.List;
+
+import static java.util.Collections.singletonList;
+import static java.util.Collections.singletonMap;
+import static java.util.Objects.requireNonNull;
+
+@Service
+@SuppressWarnings("PMD.DataflowAnomalyAnalysis")
+public class RoleAssignmentService {
+
+    private final AuthTokenGenerator serviceAuthTokenGenerator;
+
+    private final RoleAssignmentServiceApi roleAssignmentServiceApi;
+
+    private final IdamSystemTokenGenerator idamSystemTokenGenerator;
+
+    @Autowired
+    public RoleAssignmentService(RoleAssignmentServiceApi roleAssignmentServiceApi,
+                                 AuthTokenGenerator serviceAuthTokenGenerator,
+                                 IdamSystemTokenGenerator idamSystemTokenGenerator) {
+        this.roleAssignmentServiceApi = roleAssignmentServiceApi;
+        this.serviceAuthTokenGenerator = serviceAuthTokenGenerator;
+        this.idamSystemTokenGenerator = idamSystemTokenGenerator;
+    }
+
+    public List<RoleAssignment> searchRolesByCaseId(String caseId) {
+        requireNonNull(caseId, "caseId cannot be null");
+
+        RoleAssignmentResource roleAssignmentResponse = performSearch(caseId);
+
+        return roleAssignmentResponse.getRoleAssignmentResponse();
+    }
+
+
+    private RoleAssignmentResource performSearch(String caseId) {
+        try {
+            return roleAssignmentServiceApi.queryRoleAssignments(
+                idamSystemTokenGenerator.generate(),
+                serviceAuthTokenGenerator.generate(),
+                buildQueryRequest(caseId)
+            );
+        } catch (FeignException ex) {
+            throw new ServerErrorException(
+                "Could not retrieve role assignments when performing the search", ex);
```

Comment:
Probably out of the scope now, but worth mentioning. Do we want to throw exceptions every time a role is not found when searching? I don't know, but it could be acceptable for users who don't have the right permissions access the search feature. Maybe too soon to think about this now.

Reply:
Very nice spot probably something that we could review and set up a discussion on what we could do in this scenario. 
We could skip auto-assignment if this happens or fail the request definitely something to discuss.

good spot 👍